src/reference/asciidoc/xmpp.adoc

```diff
@@ -80,6 +80,42 @@ When started it will register a `PacketListener` that will listen for incoming X
 It forwards any received messages to the underlying adapter which will convert them to Spring Integration Messages and send them to the specified `channel`.
 It will unregister the `PacketListener` when it is stopped.
 
+Starting with _version 4.3_ the `ChatMessageListeningEndpoint` (and its `<int-xmpp:inbound-channel-adapter>`)
+supports a `org.jivesoftware.smack.filter.StanzaFilter` injection to be registered on the provided `XMPPConnection`
+together with an internal `StanzaListener` implementation.
+See their https://www.igniterealtime.org/builds/smack/docs/latest/javadoc/org/jivesoftware/smack/XMPPConnection.html#addAsyncStanzaListener%28org.jivesoftware.smack.StanzaListener,%20org.jivesoftware.smack.filter.StanzaFilter%29[JavaDocs] for more information.
+
+Also with the _version 4.3_ the `payload-expression` has been introduced for the `ChatMessageListeningEndpoint`.
+The incoming `org.jivesoftware.smack.packet.Message` represents a root object of evaluation context.
+This option is useful in case of <<xmpp-extensions>>.
+For example, for the GCM protocol we can extract the body using expression:
+
+[source,xml]
+----
+payload-expression="getExtension('google:mobile:data').json"
+----
+
+for the XHTML protocol:
+
+[source,xml]
+----
+payload-expression="getExtension(T(org.jivesoftware.smackx.xhtmlim.packet.XHTMLExtension).NAMESPACE).bodies[0]"
+----
+
+To simplify the access to the Extension in the XMPP Message, the `extension` variable is added into the
+`EvaluationContext`.
+Note, it is done only when one and only one Extension is present in the Message.
+The samples above with the `namespace` manipulations can be simplified to something like:
+
+[source,xml]
+ ----
+ payload-expression="#extension.json"
```

Comment:
this feature was really confusing to me until I found this part of the documentation. I would suggest to add it to the XML documentation of the payload-expression. Otherwise if may get missed.